src/triedb.rs

```diff
@@ -71,9 +76,17 @@ impl<'db> TrieDB<'db> {
     ) -> crate::Result<Option<T>> {
         match cur_node_hash {
             Some(hash) => {
-                let node_rlp = self.db.get(&hash).ok_or_else(|| TrieError::IncompleteDatabase(hash))?;
-
-                match RlpNode::decoded(&node_rlp) {
+                let node_rlp;
+                let decoded_rlp = if self.cache.borrow_mut().contains_key(&hash) {
+                    node_rlp = self.cache.borrow_mut().get_mut(&hash).unwrap().to_vec();
```

Comment:
`self.cache.borrow_mut().get_mut(&hash).unwrap().to_vec()` and `self.cache.borrow_mut().insert(hash, (&*node_rlp).to_vec())` have different return types. 

Thus, I solved the two types of conflict through `decoded_rlp`. I need your advice on this. Is it okay to implement it like this?

Reply:
```rust
                let mut cache = self.cache.borrow_mut();
                let cached_node = cache.get_mut(&hash);
                let node_rlp;
                let decoded_rlp = match cached_node {
                    Some(node_rlp) => {
                        RlpNode::decoded(node_rlp)
                    }
                    None => {
                        node_rlp = self.db.get(&hash).ok_or_else(|| TrieError::IncompleteDatabase(hash))?;
                        self.cache.borrow_mut().insert(hash, (&*node_rlp).to_vec());
                        RlpNode::decoded(&node_rlp)
                    }
                };

```
How about it?
It's a little more complicated, but the current code queries the cache twice when the data is cached.

Reply:
@sgkim126 It is a very good point, but It seems difficult to implement it as it is. This function uses recursion, then an error derives from `let mut cache = self.cache.borrow_mut();` because `borrow_mut()` duplicates. I tried to modify to solve the error, but I couldn't find a better idea.

Reply:
Okay. How about leaving a fixme comment?

Reply:
Good idea!

Reply:
Modified : )